osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -347,6 +350,10 @@ public void Filter(FilterCriteria newCriteria, bool debounce = true)
             if (newCriteria != null)
                 activeCriteria = newCriteria;
 
+            // set star rating criteria from the settings
```

Comment:
It seems that always applying the star rating criteria makes the `TestCaseBeatmapCarousel#testTraversal()` test case fail. Does anyone have a suggestion for a better place to put these lines?